var/spack/repos/builtin/packages/py-torch/package.py

```diff
@@ -114,6 +114,12 @@ class PyTorch(PythonPackage, CudaPackage, ROCmPackage):
         description="Enable breakpad crash dump library",
         when="@1.10:1.11",
     )
+    # Flash attention has very high memory requirements that may cause the build to fail
+    # https://github.com/pytorch/pytorch/issues/111526
+    # https://github.com/pytorch/pytorch/issues/124018
+    desc = "Build the flash_attention kernel for scaled dot product attention"
```

Comment:
can this just be a single definition? also not very keen on `desc` becoming a class property.

Reply:
I'm unaware of any OR in Spack's syntax, which means we need two definitions. Would `desc` -> `_desc` make you happy?

Reply:
I think the `description` kwarg shows that variant's were not really meant to be repeated. What's wrong with `conflicts("+flash_attention ~cuda ~rocm")`?

Reply:
Well, we currently routinely duplicate `variants`, including in the existing PyTorch package. Using `conflicts` doesn't really match the way CMake does things, where the variant is only defined when another variant is enabled. This is also the reason that a `when` parameter was added.

Reply:
okay then use `_*` or repeat the string value instead of using a variable.

Reply:
or `with default_args(default=True, description=...): ...` to avoid the temporary